src/pilot/assign_role.py

```diff
@@ -1049,53 +1076,101 @@ def get_drives(drac_client):
     osd_config = {
-        'osd_scenario': 'non-collocated',
-        'devices': [],
-        'dedicated_devices': []}
+        'osd_scenario': 'lvm',
+        'osd_objectstore': 'bluestore',
+        'lvm_volumes': []}
     osd_index = 0
+    vg_index = 0
     remaining_ssds = len(ssds)
-    for ssd in ssds:
+    for ssd in ssds:      
         ssd_pci_bus_number = get_pci_bus_number(ssd, controllers)
         ssd_device_name = get_by_path_device_name(ssd_pci_bus_number, ssd)
-
         num_osds_for_ssd = int(math.ceil((len(osd_drives)-osd_index) /
                                          (remaining_ssds * 1.0)))
+        # x2 volumes for each data osd (DB & WAL)
+        allocation_journals = 50 / num_osds_for_ssd - 1
+        mklvm.append('  device=$(ls -la ' + ssd_device_name + " |  awk -F \"../../\" '{ print $2 }')")
+        mklvm.append('  eval "wipefs -a /dev/${device}"')
+        mklvm.append('  sleep 2')
+        mklvm.append('  pvcreate ' + ssd_device_name)
+        mklvm.append('  vgcreate ceph_vg' + str(vg_index) + ' ' + ssd_device_name)
+        mklvm.append("  size=$(sudo fdisk -l /dev/${device} 2>/dev/null | grep -m1 \"Disk\" | awk '{print $5}')")
+        mklvm.append("  siz=`expr $((${size} * " + str(allocation_journals) + " / 100))`")
+        mklvm.append("  siz=`expr $((${siz} / 512 * 512))`")
+        for i in range(0, num_osds_for_ssd):
+            mklvm.append('  lvcreate -n ceph_lv' + str(vg_index) + "-" + str(i) + '_wal -L ${siz}B ceph_vg' + str(vg_index))
+            mklvm.append('  lvcreate -n ceph_lv' + str(vg_index) + "-" + str(i) + '_db -L ${siz}B ceph_vg' + str(vg_index))
+            mklvm.append('  sleep 2')
 
         osds_for_ssd = osd_drives[osd_index:
                                   osd_index + num_osds_for_ssd]
+        ind = 0
+        journal_index = vg_index
         for osd_drive in osds_for_ssd:
+            vg_index += 1
             osd_drive_pci_bus_number = get_pci_bus_number(osd_drive,
                                                           controllers)
             osd_drive_device_name = get_by_path_device_name(
                 osd_drive_pci_bus_number, osd_drive)
-
-            osd_config['devices'].append(osd_drive_device_name)
-            osd_config['dedicated_devices'].append(ssd_device_name)
+            mklvm.append('  device=$(ls -la ' + osd_drive_device_name + " |  awk -F \"../../\" '{ print $2 }')")
+            mklvm.append('  eval "wipefs -a /dev/${device}"')
+            mklvm.append('  sleep 2')
+            mklvm.append('  pvcreate ' + osd_drive_device_name)
+            mklvm.append('  vgcreate ceph_vg' + str(vg_index) + ' ' + osd_drive_device_name)
+            mklvm.append("  size=$(sudo fdisk -l /dev/${device} 2>/dev/null | grep -m1 \"Disk\" | awk '{print $5}')")
+            mklvm.append("  sizeb=`expr $((${size} * 99 / 100))`")
+            mklvm.append("  sizeb=`expr $((${sizeb} / 512 * 512))`")
```

Comment:
sizeb=`expr $((${sizeb} / 512 * 512))` -- isn't that always the sizeb?  why is it necessary yo divide and multiply with same number?

Reply:
The value needs to be rounded to the nearest 512 multiplier ( sector byte size ) for the volume to be created.